testsuite/tests/typecheck/should_compile/abstract_refinement_hole_fits.stderr

```diff
@@ -240,14 +240,14 @@ abstract_refinement_hole_fits.hs:7:5: warning: [-Wtyped-holes (in -Wdefault)]
           where const :: forall a b. a -> b -> a
         ($) (_ :: a5 -> Integer -> [Integer] -> Integer) (_ :: a5)
           where ($) :: forall a b. (a -> b) -> a -> b
+        uncurry (_ :: a4 -> b3 -> Integer -> [Integer] -> Integer)
+                (_ :: (a4, b3))
+          where uncurry :: forall a b c. (a -> b -> c) -> (a, b) -> c
+        ($!) (_ :: a6 -> Integer -> [Integer] -> Integer) (_ :: a6)
+          where ($!) :: forall a b. (a -> b) -> a -> b
         fail (_ :: [Char]) (_ :: t1)
           where fail :: forall (m :: * -> *) a. Monad m => String -> m a
         return (_ :: Integer -> [Integer] -> Integer) (_ :: t1)
           where return :: forall (m :: * -> *) a. Monad m => a -> m a
         pure (_ :: Integer -> [Integer] -> Integer) (_ :: t1)
           where pure :: forall (f :: * -> *) a. Applicative f => a -> f a
-        uncurry (_ :: a4 -> b3 -> Integer -> [Integer] -> Integer)
-                (_ :: (a4, b3))
-          where uncurry :: forall a b c. (a -> b -> c) -> (a, b) -> c
-        ($!) (_ :: a6 -> Integer -> [Integer] -> Integer) (_ :: a6)
-          where ($!) :: forall a b. (a -> b) -> a -> b
```

Comment:
This whole file (and the following few) looks unrelated (also weird!)

Reply:
The ordering in hole tests seems to vary depending on internals, and it seems that it changed together with `(->)`. Ideally, it should be sorted, but I don't want to change this in this PR.

Reply:
Absolutely.